File: php/all-your-base/AllYourBase.php
<?php declare(strict_types=1);

function rebase(int $fromBase, array $digits, int $toBase): array
{
    if ($fromBase < 2) {
        throw new InvalidArgumentException("input base must be >= 2");
    }

    if ($toBase < 2) {
        throw new InvalidArgumentException("output base must be >= 2");
    }
    $baseTen = 0;

    foreach(array_reverse($digits) as $index => $digit) {
        if ($digit < 0 || $digit >= $fromBase) {
            throw new InvalidArgumentException("all digits must satisfy 0 <= d < input base");
        }
        $realNum = $digit * ($fromBase ** ($index));
        $baseTen += $realNum;
    }

    $result = [];
    $counter = 1000;
    while ($baseTen >= $toBase) {
        $remainder = $baseTen % $toBase;
        $baseTen = (int)floor($baseTen / $toBase);
        $result[] = $remainder;
        $counter--;
        if ($counter < 1){
            break;
        }
    }

    $result[] = $baseTen % $toBase;

    return array_reverse($result);
}

File: python/all-your-base/all_your_base.py
def rebase(input_base, digits, output_base):
    if input_base < 2:
        raise ValueError("input base must be >= 2")

    if output_base < 2:
        raise ValueError("output base must be >= 2")
    base = 0
    for i, v in enumerate(reversed(digits)):
        if v >= input_base or v < 0:
            raise ValueError("all digits must satisfy 0 <= d < input base")
        base += v * input_base ** i

    if base == 0:
        return [0]

    result = []
    power = 0

    while True:
        if output_base ** (power + 1) > base:
            break
        power += 1

    for p in range(power, -1, -1):
        high = output_base ** p
        i = 0
        while True:
            if high * (i + 1) > base:
                break
            i += 1

        result.append(i)
        base -= high * i

    return result